### PR TITLE
sshified: init at 1.2.1

### PR DESCRIPTION
**Pull request by joinemm**
https://github.com/hoffie/sshified is a http proxy that tunnels requests through ssh connections.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/by-name/ss/sshified/package.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  fetchFromGitHub,
+}:
+
+buildGoModule rec {
+  pname = "sshified";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "hoffie";
+    repo = pname;
```
Comment (Aleksanaa) [inline pkgs/by-name/ss/sshified/package.nix:13]: nit: may not be necessary but
```suggestion
    repo = "sshified";
```

File: pkgs/by-name/ss/sshified/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    rev = "v${version}";
```
Comment (fabaff) [inline pkgs/by-name/ss/sshified/package.nix:14]: Helps with branch/tag confusion.

```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/by-name/ss/sshified/package.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    hash = "sha256-oCeuQ4Do+Lyqsf8hBH9qvLxWbWQlqol481VrbnAW2ic=";
+  };
+
+  vendorHash = null;
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.Version=${version}"
```
Comment (fabaff) [inline pkgs/by-name/ss/sshified/package.nix:23]:
```suggestion
    "-X=main.Version=${version}"
```

File: pkgs/by-name/ss/sshified/package.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  ];
+
+  subPackages = [ "." ];
+
+  meta = {
+    description = "Proxy HTTP requests through SSH";
+    homepage = "https://github.com/hoffie/sshified";
```
Comment (fabaff) [inline pkgs/by-name/ss/sshified/package.nix:30]:
```suggestion
    homepage = "https://github.com/hoffie/sshified";
    changelog = "https://github.com/hoffie/sshified/blob/v${version}/CHANGELOG.md";
```

**Comment by joinemm** [comment]
@fabaff @Aleksanaa Thanks, added your suggested changes.

**Review by Aleksanaa — APPROVED** [review]

**Comment by Aleksanaa** [comment]
built on aarch64-darwin

**Pull request merged**